oeps/oep-0019-bp-developer-documentation.rst

```diff
@@ -0,0 +1,298 @@
+
+========================
+rST Questions / Comments
+========================
+
+* rST Formatting Questions:
+
+   * [Robert] I like the header format that Nimisha uses elsewhere that just uses an underline, rather than a line above
+      and below. I think it has more overhead to do above and below, but I think the doc team uses this. Could we update
+      templates to match our preference?
+
+   * Do we need to keep to 80 characters per line? What tools will make this easy?
+
+      * Could try `Insert Break After Wrapping Width 80 characters in Sublime Text 3`_
+
+* What tools can help with rST?
+
+   * PyCharm under development will have features, but it is broken for other reasons.
+
+   * Nimisha uses VisualStudio Code - `Restructured Text Previewer`_.
+
+   * `Online reStructured Text editor`_
+
+   * `Online table generator`_ (supports multiple formats)
+
+   * `Pandoc`_ (a universal document converter).  This can be used to convert between many formats, like Google Doc to
+      rST.
+
+     code::
+
+         pandoc example_source.docx -f docx -t rst -s -o example_output.rst
+
+   * Note: It looks like I got a section like this started below.
+
+.. _Insert Break After Wrapping Width 80 characters in Sublime Text 3: https://stackoverflow.com/questions/32362277/insert-break-after-wrapping-width-80-characters-in-sublime-text-3
+.. _Restructured Text Previewer: https://marketplace.visualstudio.com/items?itemName=tht13.rst-vscode
+.. _Online reStructured Text editor: http://rst.ninjs.org/
+.. _Online table generator: http://truben.no/table/
+.. _Pandoc: https://pandoc.org/
+
+********
+Abstract
+********
+
+Developers who work on the Open edX platform, both inside and outside of edX,
+need accurate and current documentation about platform architecture, APIs, and
+development best practices.
+
+**********
+Motivation
+**********
+
+Developer documentation has been inconsistently and in some cases poorly
+maintained. Documents exist on ReadTheDocs (RTD), in repositories on GitHub,
+in Confluence, and in Google Docs. Documents are also difficult to find.
+
+*************
+Specification
+*************
+
+===========================
+Documentation Format in rST
+===========================
+
+Documentation that is added to a Git repository will use the reStructuredText
+(rST) format.
+
+=======================
+rST Tools and Resources
+=======================
+
+* `reStructuredText (rST) Primer`_
+
+* `Online rST editor`_
+
+* `Pandoc: a command line converter to and from rST`_
+
+* Blog post about `why rST over Markdown for documentation`_.
+
+`reStructuredText (rST) Primer`: http://www.sphinx-doc.org/en/master/usage/restructuredtext/basics.html
+`Online rST editor`: http://rst.ninjs.org/
+`Pandoc: a command line converter to and from rST`: https://pandoc.org/
+`why rST over Markdown for documentation`: http://ericholscher.com/blog/2016/mar/15/dont-use-markdown-for-technical-docs/
```

Comment:
+1 for Markdown. It's the lingua franca for technical documentation in open source communities, and MkDocs may not have all the bells and whistles of rST but it does a pretty good job. https://www.mkdocs.org/

Don't get me wrong - I love ReadTheDocs project and applaud everything that Eric Holscher is doing to improve documentation of open source projects. I'm just not sure that rST is the best choice if you want to get community contributions of documentation.

Reply:
I'll have to loop in @nedbat on this.  I believe the authors of this OEP don't feel as strongly about this particular point.  However, we based this particular decision on input we received from @nedbat and his myriad past experiences and issues with not just using rST from the start.

Reply:
I would like to +1 this as well.  I prefer Markdown to RST for its speed and clarity.

Reply:
* Restructured Text support inline links if you prefer them, though out-of-band often makes reading the prose easier because it keeps the URLs out of the way.
* If a `:ref:` is in the text, it's because we want to make a reference to something else, which Markdown simply can't do. So either use the ref because we want the link, or leave it out because the text should be readable.  Choosing a more limited format in order to preclude the option seems like an odd strategy.
* Rst is the lingua-franca in Python documentation.
* I am certain that anyone interested can learn the Rst way of doing whatever Markdown can do. They are simply not that different for the limited set of features that Markdown supports.  But .rst gives us the features where we need them, for example in multi-section documentation.